lib/jwt.rb

```diff
@@ -163,17 +161,17 @@ def decode(jwt, key=nil, verify=true, options={}, &keyfinder)
     if options[:verify_not_before] && payload.include?('nbf')
       raise JWT::ImmatureSignature.new('Signature nbf has not been reached') unless payload['nbf'].to_i < (Time.now.to_i + options[:leeway])
     end
-    if options[:verify_iss] && payload.include?('iss')
-      raise JWT::InvalidIssuerError.new("Invalid issuer. Expected #{options['iss']}, received #{payload['iss']}") unless payload['iss'].to_s == options['iss'].to_s
+    if options[:verify_iss] && options['iss']
```

Comment:
Thanks for the review :)

Going to add more tests and implement checks for these cases.